partcad/src/partcad/runtime_python.py

```
@@ -280,16 +315,94 @@
+        guard_path = os.path.join(path, ".partcad.installed." + python_package_hash)
+        if session:
+            # Add the dependency to the session dependencies
+            session["deps"].append(python_package)
+            if not os.path.exists(guard_path):
+                # Mark this session as needed if the dependency is not met by the runtime environment
+                session["dirty"] = True
+        else:
+            with self.sync_lock():
+                if not os.path.exists(guard_path):
+                    item = python_package
+                    if not path is None:
```

Ruff (0.8.2)

Test for object identity should be `is not`

Convert to `is not`

(E714)